extensions/cassandra-storage/src/main/java/io/druid/storage/cassandra/CassandraLoadSpec.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Druid - a distributed column store.
+ * Copyright 2012 - 2015 Metamarkets Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.druid.storage.cassandra;
+
+import com.fasterxml.jackson.annotation.JacksonInject;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.metamx.common.ISE;
+import io.druid.segment.loading.DataSegmentPuller;
+import io.druid.segment.loading.LoadSpec;
+import io.druid.segment.loading.SegmentLoadingException;
+
+import java.io.File;
+import java.util.Map;
+
+/**
+ *
+ */
+@JsonTypeName(CassandraDataSegmentPuller.SCHEME)
+public class CassandraLoadSpec implements LoadSpec
+{
+  @JsonProperty
+  private final String key;
+  private final CassandraDataSegmentPuller puller;
+  @JsonCreator
+  public CassandraLoadSpec(
+      @JacksonInject Map<String, DataSegmentPuller> pullers,
```

Comment:
The LoadSpec is stored as a Map until the OmniSegmentLoader is called to load the segment. At that point, the LoadSpec's respective modules have to be loaded anyways, so the mapper injection can go away in favor of the explicit puller injection.